src/mongo_client.ts

```diff
@@ -536,6 +538,10 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> implements
    * @see docs.mongodb.org/manual/reference/connection-string/
    */
   async connect(): Promise<this> {
+    if (this.closeLock) {
```

Comment:
I don't know that we came to consensus about the best way to handle concurrent closes / opens.  How would you feel about reverting this change and adding a tech chat item to discuss or filing a ticket to address it?  We don't need to fix it now, this is a problem we've always had.

Reply:
Good call. I pushed this up because was going to bring this up in tech chat, but I think it slipped my mind. I'll revert it and file a follow-up ticket!

Reply:
https://jira.mongodb.org/browse/NODE-6691